### ISSUE TITLE
Mozilla removed the add-on?

### ISSUE THREAD

**Issue by vvug** [iamadamdev/bypass-paywalls-firefox#82]
The https://addons.mozilla.org/en-US/firefox/addon/bypasspaywalls/ URL from `README.md` returns: Page not found.

**Comment by iamadamdev** [comment]
One of the community reviewers removed it from the store claiming it violated the ToS, which is BS. The ToS never mentions the word paywall once. Please support me in appealing! https://wiki.mozilla.org/Add-ons#Getting_in_touch

**Comment by Thomasc5** [comment]
> 
> 
> One of the community reviewers removed it from the store claiming it violated the ToS, which is BS. The ToS never mentions the word paywall once. Please support me in appealing! https://wiki.mozilla.org/Add-ons#Getting_in_touch

May I ask if it is possible to extract the add-on from Firefox on my old PC and install it on new PC and Android phone?  Thank you.

**Comment by jrwhite** [comment]
[iamadamdev@hotmail.com.xpi.txt](https://github.com/iamadamdev/bypass-paywalls-firefox/files/2609079/iamadamdev.hotmail.com.xpi.txt)
Here is the .xpi file. Github doesn't allow uploading .xpi files so i added a .txt extension. Remove that after you download it. To install on firefox, naviagte to the .xpi file, right click, 'open with' firefox.

**Comment by kilgarenone** [comment]
Oh man, I read alot of FT. I need this.

Which channel in here  https://wiki.mozilla.org/Add-ons#Getting_in_touch is the most effective to appeal to?

**Comment by Nicholas-Steel** [comment]
Thank you jrwhite that installed without issue.

**Comment by esteves023** [comment]
> 
> 
> [iamadamdev@hotmail.com.xpi.txt](https://github.com/iamadamdev/bypass-paywalls-firefox/files/2609079/iamadamdev.hotmail.com.xpi.txt)
> Here is the .xpi file. Github doesn't allow uploading .xpi files so i added a .txt extension. Remove that after you download it. To install on firefox, naviagte to the .xpi file, right click, 'open with' firefox.

Do you have the file for chrome too? Couldn't do the thing for chrome.

**Comment by ohadcn** [comment]
@iamadamdev meanwhile, can you upload the addon to [github releases page](https://github.com/iamadamdev/bypass-paywalls-firefox/releases) so people would be able to download it from there? and add a link in readme.md?

**Comment by gjmbeyond** [comment]
I am using Firefox. I have download the ZIP file, how can I install this extension in my Firefox Browser?  Need some help.

**Comment by jrwhite** [comment]
@gjmbeyond did you try using the method in my response above?

**Comment by gjmbeyond** [comment]
> 
> 
> @gjmbeyond did you try using the method in my response above?

OK, thank you very much! Installed successfully!
But here is another problem, FT works perfectly, but WSJ still need to subscribe for the whole text. I have removed all 'adblock' like extensions, still doesn't work.

**Comment by Nicholas-Steel** [comment]
> 
> 
> [iamadamdev@hotmail.com.xpi.txt](https://github.com/iamadamdev/bypass-paywalls-firefox/files/2609079/iamadamdev.hotmail.com.xpi.txt)
> Here is the .xpi file. Github doesn't allow uploading .xpi files so i added a .txt extension. Remove that after you download it. To install on firefox, naviagte to the .xpi file, right click, 'open with' firefox.

Is this based on the newest code? There appears to have been a fix for WSJ websites about 3 weeks ago.

**Comment by gjmbeyond** [comment]
Yes, based on @jrwhite ‘s txt 4 days ago. Neither my Chrome nor Firefox works for WSJ.

**Comment by neuhaus** [comment]
Interesteting comment by user jape on [HN](https://news.ycombinator.com/item?id=18567578):

> 
> While this is extremely dubious behavior from Mozilla, and reminds me why I stopped donating to them (the moment Firefox started requiring _their_ signature in order to load addons), Mozilla still has the "automated signing API" in place. Supposedly, this API allows to get an XPI signed as long as it passes a series of automated checks. So it's worth a try.
> 
> This was the excuse they used anyway when trying to justify their signature requirements were "not a walled garden". I didn't believe it of course.
> 
> You can also just mark the addon as "not listed in AMO" when submitting it to addons.mozilla.org and it will not be listed on the store, but it will be signed. More details in https://developer.mozilla.org/docs/Mozilla/Add-ons/Distribution

**Comment by andreicristianpetcu** [comment]
[Click here](https://github.com/andreicristianpetcu/bypass-paywalls-firefox/releases/download/1.4.3/bypass_paywalls-1.4.3-an+fx.xpi) and it will get installed. It's signed.

**Comment by andreicristianpetcu** [comment]
I opened an issue showing how to distribute the signed addon here https://github.com/iamadamdev/bypass-paywalls-firefox/issues/86

**Comment by BillDorn** [comment]
@andreicristianpetcu 
It says "All the signing options are subject to the Firefox Add-on Distribution Agreement." [here](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/Distribution)
So does self-hosting the xpi really solve the issue?

**Comment by andreicristianpetcu** [comment]
we shall see if my forked paywall addon gets banned in a week or two or not

**Comment by championshuttler** [comment]
> Oh man, I read alot of FT. I need this.
> 
> Which channel in here https://wiki.mozilla.org/Add-ons#Getting_in_touch is the most effective to appeal to?

Hi @kilgarenone you can react out to addons reviewers in `#addon-reviewers` channel

**Comment by championshuttler** [comment]
Apart from that you can send an email to `amo-admins AT mozilla DOT org`
